### ISSUE TITLE
While updating 1.17 an ERROR

### ISSUE THREAD

**Issue by SwiftExplorer567** [ModsByLeo/SpeedTrading#5]
Can anybody help me to fix this problem?

```
---- Minecraft Crash Report ----
// Shall we play a game?

Time: 18.01.2022 11:15
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'speedtrading'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:457)
	at net.minecraft.client.main.Main.main(Main.java:179)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NoClassDefFoundError: adudecalledleo/lionutils/LoggerUtil
	at adudecalledleo.speedtrading.SpeedTrading.<clinit>(SpeedTrading.java:16)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:466)
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
	at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 11 more
Caused by: java.lang.ClassNotFoundException: adudecalledleo.lionutils.LoggerUtil
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:158)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
	... 18 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at net.minecraft.class_310.<init>(class_310.java:457)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:179)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.17.1
	Minecraft Version ID: 1.17.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 965913560 bytes (921 MiB) / 1610612736 bytes (1536 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 6
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 3500X 6-Core Processor             
	Identifier: AuthenticAMD Family 23 Model 113 Stepping 0
	Microarchitecture: Zen 2
	Frequency (GHz): 3,59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 6
	Graphics card #0 name: NVIDIA GeForce RTX 2060
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095,00
	Graphics card #0 deviceId: 0x1e89
	Graphics card #0 versionInfo: DriverVersion=30.0.15.1123
	Memory slot #0 capacity (MB): 8192,00
	Memory slot #0 clockSpeed (GHz): 3,00
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 8192,00
	Memory slot #1 clockSpeed (GHz): 3,00
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 22229,18
	Virtual memory used (MB): 9924,84
	Swap memory total (MB): 5896,50
	Swap memory used (MB): 27,58
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		autoconfig1u: Auto Config v1 Updated 3.2.2
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-config2: Cloth Config v4 4.8.1
		fabric: Fabric API 0.46.1+1.17
		fabric-api-base: Fabric API Base 0.4.0+cf39a74318
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.5.0+e821752d18
		fabric-biome-api-v1: Fabric Biome API (v1) 3.2.2+cf39a74318
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.6+cf39a74318
		fabric-command-api-v1: Fabric Command API (v1) 1.1.4+cf39a74318
		fabric-commands-v0: Fabric Commands (v0) 0.2.3+cf39a74318
		fabric-containers-v0: Fabric Containers (v0) 0.1.13+cf39a74318
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.4.1+3447790d18
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.6+cf39a74318
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.15+b556f28c18
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.1+377137cc18
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.12+e99fbe1218
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.4+cf39a74318
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.8+cf39a74318
		fabric-item-api-v1: Fabric Item API (v1) 1.3.0+6617390918
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.2+cf39a74318
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.6+2a2bb57318
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.4+cf39a74318
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.6+0392f3a618
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.5+cf39a74318
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 1.0.3+cf39a74318
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.7+cf39a74318
		fabric-models-v0: Fabric Models (v0) 0.3.1+cf39a74318
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.14+cf39a74318
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.12+cf39a74318
		fabric-networking-v0: Fabric Networking (v0) 0.3.3+cf39a74318
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.11.1+f907116918
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.8+cf39a74318
		fabric-particles-v1: Fabric Particles (v1) 0.2.5+cf39a74318
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.8.0+ea29b33318
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.6+ce16ae5918
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.10+ce16ae5918
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.6+cf39a74318
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.6+cf39a74318
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.2.1+cf39a74318
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.7+cf39a74318
		fabric-rendering-v1: Fabric Rendering (v1) 1.10.1+377137cc18
		fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 1.0.1+591a0f2118
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.10+f09604ce18
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.5+cf39a74318
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.9+cf39a74318
		fabric-structure-api-v1: Fabric Structure API (v1) 1.2.0+48a9ad8918
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.2.3+cf39a74318
		fabric-textures-v0: Fabric Textures (v0) 1.0.7+cf39a74318
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.3.3+3b96517518
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.4+cf39a74318
		fabricloader: Fabric Loader 0.12.12
		java: OpenJDK 64-Bit Server VM 16
		minecraft: Minecraft 1.17.1
		speedtrading: Speed Trading 2.0.1
	Launched Version: fabric-loader-0.12.12-1.17.1
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: NO CONTEXT
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>
```

**Comment by Leo40Git** [comment]
1.17 isn't currently supported, and won't be.

_However_, updating the mod to 1.18 would also pretty much require me to do the 1.16 -> 1.17 changes anyway, so I'll release a version when I'm done doing that.

By the way, the 1.16 versions cannot run on Minecraft 1.17, even if you patch out uses of my old deprecated library.

**Comment by Leo40Git** [comment]
v2.1.0 is now available for 1.17 and 1.18.
